DOCSP-36999 - Databases and Collections (#24)
mongodb/docs-pymongo: source/fundamentals/databases-and-collections.txt

In the following code example, the tag set passed to the ``read_preference`` argument
instructs {+driver-short+} to prefer reads from the
New York data center (``dc``), but to fall back to the San Francisco data center:

.. code-block:: python

   >>> from pymongo.read_preferences import Secondary
   >>> db = client.get_database(
   ...     'test', read_preference=Secondary([{'dc': 'ny'}, {'dc': 'sf'}]))
   >>> db.read_preference
   Secondary(tag_sets=[{'dc': 'ny'}, {'dc': 'sf'}])

When performing a read, the ``MongoClient`` in the preceding example tries to find
secondary members in New York, then in San Francisco. If it can't find a secondary member
in either location, it raises an ``AutoReconnect`` error.

.. tip::

   To avoid an ``AutoReconnect`` error, include an empty dictionary (``{}``) at the end of
   the list. This instructs {+driver-short+} to read from any member that matches the mode,
   ignoring tags.

.. _distributes reads to secondaries:

Local Threshold
~~~~~~~~~~~~~~~

If multiple replica-set members match the read preference and tag sets you specify,
{+driver-short+} reads from among the nearest members, chosen according to ping time.

> By default, only members whose ping times are within 15 milliseconds of the nearest
> are used for queries. To distribute reads among members with

Reviewer: S: It took me a while to parse what this sentence was trying to say

```suggestion
By default, the driver only uses those whose ping times are within 15 milliseconds of the nearest members for queries.
```

higher latencies, pass the ``localThresholdMS`` to the ``MongoClient()`` constructor.

The following example specifies a local threshold of 35 milliseconds:

.. code-block:: python

   >>> client = pymongo.MongoClient(
   ...     replicaSet='repl0',
   ...     readPreference='secondaryPreferred',
   ...     localThresholdMS=35)

In the preceding example, {+driver-short+} distributes reads among matching members
within 35 milliseconds of the closest member's ping time.

.. note::
  
  {+driver-short+} ignores the value of ``localThresholdMS`` when communicating with a
  replica set through a ``mongos`` instance. In this case, use the

>   :manual:`localThreshold </reference/program/mongos/#std-option-mongos.--localThreshold>` command-
>   line option.

Reviewer: The line break here is causing a space between command- and line
```suggestion
  :manual:`localThreshold </reference/program/mongos/#std-option-mongos.--localThreshold>` 
  command-line option.
```
